### PR TITLE
ddl: use `advertise-addr` as the instance address

### LINKED ISSUE #43983: Checkpoint is unexpectedly used when DDL owner is changed
## Bug Report

Please answer these questions before submitting your issue. Thanks!

### 1. Minimal reproduce step (Required)

1. Create index for a large table.
2. Kill TiDB DDL owner.

### 2. What did you expect to see? (Required)

Starting from the global checkpoint.

### 3. What did you see instead (Required)

Starting from the local checkpoint.

### 4. What is your TiDB version? (Required)

6e5604f08

### PR DESCRIPTION
**Pull request by tangenta**
### What problem does this PR solve?

Issue Number: close #43983

Problem Summary:

When DDL owner is changed, the local checkpoint is mistakenly considered to be valid because the instance addresses are the same.

```
[checkpoint.go:289] ["[ddl-ingest] resume checkpoint"] ["job ID"=127] ["index ID"=30] ["local checkpoint"=74800000000000xxx] ["global checkpoint"=] ["previous instance"=[0.0.0.0:4000](http://0.0.0.0:4000/):/data] ["current instance"=[0.0.0.0:4000](http://0.0.0.0:4000/):/data]
```

### What is changed and how it works?

- Use "advertise-address" to identify the TiDB instance.
- Fix the issue that the checkpoint may take no effect in partition tables.

### Check List

Tests 

- [ ] Unit test
- [ ] Integration test
- [x] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

File: ddl/ingest/checkpoint.go [hunk 1]
```diff
@@ -300,6 +291,22 @@ func (s *CheckpointManager) resumeCheckpoint() error {
 	})
 }
 
+func getCurrentInstanceAddr(ctx context.Context, se *sess.Session, serverID uint64) string {
+	template := "select INSTANCE from INFORMATION_SCHEMA.CLUSTER_INFO where SERVER_ID = %d;"
```
Comment (wjhuang2016) [inline ddl/ingest/checkpoint.go:295]: It can't get the instance info if SEM is enabled

Comment (tangenta) [reply ddl/ingest/checkpoint.go]: Updated. Now we use `config.advertise-address` instead.

**Review by Benjamin2037 — APPROVED** [review]
LGTM

**Pull request merged**